FWCore/ParameterSet/src/ProcessDesc.cc

```diff
@@ -30,14 +29,13 @@ namespace edm {
   }
 
   void ProcessDesc::addDefaultService(std::string const& service) {
-    typedef std::vector<ParameterSet>::iterator Iter;
-    for(Iter it = services_->begin(), itEnd = services_->end(); it != itEnd; ++it) {
+    for(auto it = services_.begin(), itEnd = services_.end(); it != itEnd; ++it) {
       std::string name = it->getParameter<std::string>("@service_type");
       if (name == service) {
         // Use the configured service.  Don't add a default.
         // However, the service needs to be moved to the front because it is a standard service.
         ParameterSet pset = *it;
-        services_->erase(it);
+        services_.erase(it);
```

Comment:
Well, the `erase` was in there before I tweaked it.  And yes, in general, it does invalidate the iterator.  But in this case, and the following, the loop is exited immediately after the erase (either by `return` or `break`); it's therefore safe in these cases.